### ISSUE TITLE
some molecules fail to generate BDE values in the pre-trained model, but works in the web API

### ISSUE THREAD

**Issue by LuckyLittleMonster** [NREL/alfabet#12]
Empty DataFrame
Columns: [bond_type, fragment1, fragment2, is_valid_stereo, bde_pred, bdfe_pred, is_valid, molecule, bond_index, bde, bdfe, set]
Index: []

**Comment by pstjohn** [comment]
do you have an example for a molecule this happens for?

**Comment by denfromufa** [comment]
@pstjohn is it possible to add more logging to Alfabet to find out why empty result is returned?

**Comment by pstjohn** [comment]
Possibly, are you not able to share an example molecule that gives an empty output?

**Comment by LuckyLittleMonster** [comment]
I found 3 example molecules. 

CC(=O)OCC1=C\CC/C(C)=C/CC[C@@]2(C)CC[C@@](C(C)C)(/C=C/1)O2
CC1=C\C/C=C(\C)CC[C@H]2C(C)(C)[C@@H](\C=C/1)CC[C@]2(C)O
CN1C[C@H]2CN(C)C[C@H](C1)[C@]2(O)c1ccccc1

-----------------------------------------------------------------
>>> mol = "CC(=O)OCC1=C\CC/C(C)=C/CC[C@@]2(C)CC[C@@](C(C)C)(/C=C/1)O2"
>>> model.predict([mol])
1/1 [==============================] - 6s 6s/step
Empty DataFrame
Columns: [bond_type, fragment1, fragment2, is_valid_stereo, bde_pred, bdfe_pred, is_valid, molecule, bond_index, bde, bdfe, set]
Index: []
>>> mol2 = "CC1=C\C/C=C(\C)CC[C@H]2C(C)(C)[C@@H](\C=C/1)CC[C@]2(C)O"
>>> model.predict([mol2])
1/1 [==============================] - 0s 123ms/step
Empty DataFrame
Columns: [bond_type, fragment1, fragment2, is_valid_stereo, bde_pred, bdfe_pred, is_valid, molecule, bond_index, bde, bdfe, set]
Index: []
>>> mol3 = "CN1C[C@H]2CN(C)C[C@H](C1)[C@]2(O)c1ccccc1"
>>> model.predict([mol3])
1/1 [==============================] - 0s 126ms/step
Empty DataFrame
Columns: [bond_type, fragment1, fragment2, is_valid_stereo, bde_pred, bdfe_pred, is_valid, molecule, bond_index, bde, bdfe, set]
Index: []

**Comment by pstjohn** [comment]
Where did you get those SMILES strings? I'm getting rdkit parse errors:

```python
>>> rdkit.Chem.MolFromSmiles('CC(=O)OCC1=C\CC/C(C)=C/CC[C@@]2(C)CCC@@(/C=C/1)O2')
[13:07:52] SMILES Parse Error: syntax error while parsing: CC(=O)OCC1=C\CC/C(C)=C/CC[C@@]2(C)CCC@@(/C=C/1)O2
[13:07:52] SMILES Parse Error: Failed parsing SMILES 'CC(=O)OCC1=C\CC/C(C)=C/CC[C@@]2(C)CCC@@(/C=C/1)O2' for input: 'CC(=O)OCC1=C\CC/C(C)=C/CC[C@@]2(C)CCC@@(/C=C/1)O2'
```

https://bde.ml.nrel.gov/result?name=%27CC%28%3DO%29OCC1%3DC%5CCC%2FC%28C%29%3DC%2FCC%5BC%40%40%5D2%28C%29CCC%40%40%28%2FC%3DC%2F1%29O2

I'm not sure why alfabet isn't raising those errors directly, that's probably something I can fix

**Comment by LuckyLittleMonster** [comment]
The SMILES strings are in ChEMBL database.
https://ftp.ebi.ac.uk/pub/databases/chembl/ChEMBLdb/latest/chembl_31_chemreps.txt.gz

It seems that there are some errors while I copy the smiles string to github webpage. Because there "@" in the strings. 
Empty smiles strings:
`CC(=O)OCC1=C\CC/C(C)=C/CC[C@@]2(C)CC[C@@](C(C)C)(/C=C/1)O2`
`CC1=C\C/C=C(\C)CC[C@H]2C(C)(C)[C@@H](\C=C/1)CC[C@]2(C)O`
`CN1C[C@H]2CN(C)C[C@H](C1)[C@]2(O)c1ccccc1`

The webpage (e.g. https://bde.ml.nrel.gov/result?name=CN1C%5BC%40H%5D2CN%28C%29C%5BC%40H%5D%28C1%29%5BC%40%40%5D2%28O%29c1ccccc1) will give the result but the python will return empty dataframe.

**Comment by pstjohn** [comment]
Thanks for this, I see the issue now. I likely introduced this is 0.4.0; I have a fix and test (i think) in #13

**Comment by pstjohn** [comment]
Releasing a new patch version 0.4.1 to handle this -- let me know if you still have issues
